js/misc/paygManager.js

```diff
@@ -88,6 +110,27 @@ function _isPaygEnabled() {
     return false;
 }
 
+// Takes an UNIX timestamp (in seconds) and returns a string
+// with a precision level appropriate to show to the user.
+function timeToString(seconds) {
+    if (seconds < 60)
+        return Gettext.ngettext("%s second", "%s seconds", seconds).format(seconds);
+
+    let minutes = Math.ceil(seconds / 60);
```

Comment:
Actually, I see that we explicitly call with the round number of target seconds, so we should do the proper rounding and use floor here, especially since when we call this upon logging in, it won't be with the target number of seconds but with the actual time remaining, so we do need to display an amount of time that does not exceed what is actually left.

Reply:
You're right. I used ceiling because I initially had a different idea of what I ended up implementing, but I agree it should be `Math.floor()` now.

Reply:
Actually, we can't plainly use `floor()` either in all cases, as I've realized while testing the latest changes I just made, which show an issue when you **unlock the machine from outside a session**:

Imagine the scenario when you boot up to a locked machine and unlock it with a code for 2 days, and then you log in to the recently-unlocked system: at the time the shell starts the timestamp that will be checked to decide whether to show a notification to the user with the time remaining will already be different (more recent) than the one used when unlocking the machine, meaning that the string formatting logic (that uses `floor()` to calculate the number of minutes, hours and days) will makes up end up showing a message like "PAYG time will expire in 1 day" to the user, which would be very confusing since you just entered a code for 2 days!

In the same line, if you enter a code for 1 day, you'll start to a notification saying "PAYG time will expire in 23 hours", which is not as bad but still a bit confusing, although in that case I think it cold be much less confusing than in the case of 2 vs 1 days, although I still think some correction might be needed there as well.

Last, if you enter a code for 1 hour, you'll boot up to a message that says "PAYG time will expire in 59 minutes", which I don't see any trouble with in terms of causing confusion.

TL;DR: I think we should add some kind of epsilon before using `floor()` directly for some of the biggest units, to avoid confusion. In particular, I think we should definitely use an EPSILON for the case of days because it's **very** confusing to unlock with a code for 2 days and see right away that you only have one left. For strings formatted in hours ("23 hours left" if you just unlocked for 1 day), I'm not really that concerned but I think some adjustment would be useful too.

Reply:
On a second thought, I'm not entirely happy with this solution either, since it doesn't really removes potential confusion from the user in cases other than "logging in just (in less than 5 min) after unlocking the machine. For instance, imagine you unlock the machine with a code for 2 days but take more than 5 min to log-in for any reason: in that case we'd be still showing "1 day left" which might be equally confusing. Or maybe you log in fast enough but then reboot a few minutes later and you see "1 day left".... confusing.

So, I now think a better solution might actually be something not involving epsilons, but simply reporting the actual amount of time with a bit more of detail, that is: returning a string "in seconds" for times under 1 minute, a string "in minutes" for times under 2 hours, "in hours and minutes" (if applicable) for times under 1 day, and then "in days and hours" (if applicable) for anything longer than that in days.

I've tested this and I think that's the approach that works best, and seeing  "1 day 23 hours left" right after unlocking with a code for 2 days should not be much of a problem, while way less confusing than just "flooring" it to "1 day". Also, note that the periodic reminders will be always round numbers (e.g. "2 days", "1 hour"...) because we define them that way, and so the more detailed format will only be shown as a one-off notification on startup (where you would expect a bit more of precision),  so I think it should be fine.

Reply:
Another possibility is to avoid the crazy rounding altogether and display something like the following...

Your code will expire in:
XX day(s) YY hour(s) ZZ minute(s)

Reply:
Yes, I thought of that too. But the problem is that we define the periodic notifications with such values that you would print many "zero values" in there (e.g. "0 days, 2 hours 0 minutes"), so some formatting is necessary.